project-koku/nise
# Add currency support

Adding the ability to pass in currency argument -c where you can pass in a currency code to get reports in foreign currencies to test MVP currency.

Testing instructions:
1. checkout and pull branch
2. Create reports for AWS, Azure, and GCP
```
AWS report:
nise report -c jpy aws --static-report-file ./example_aws_static_data.yml --aws-s3-report-name <directory_name> --aws-s3-bucket-name ./

Azure Report:
nise report -c nok azure --static-report-file ./example_azure_static_data.yml --azure-report-name <directory_name> --azure-container-name ./

GCP Report:
nise report -c jpy gcp -s 2021-01-01 -e 2021-01-02  --gcp-report-prefix <report_name> --gcp-bucket-name ./  

```
3. check reports for currency to see if it matches in passed in value, valid currency options below
```
valid_currencies = [
        "aud",
        "cad",
        "chf",
        "cny",
        "dkk",
        "eur",
        "gbp",
        "hkd",
        "jpy",
        "nok",
        "nzd",
        "sek",
        "sgd",
        "usd",
        "zar",
    ]
```
4. If the currency code is not in support list
```
nise report: error: argument -c/--currency: nkd is an unsupported currency code.
```
5. If no code is given should default currency to USD
